File: L18-Intro-to-Ruby/10-blocks.js
function higherOrderFunction(callback) {
  console.log('Starting...');
  callback()
  console.log('Finished!');
}

higherOrderFunction(() => {
  console.log('HELLO FROM ARROW FUNCTION!');
})

higherOrderFunction(function(){
  console.log('hello from regular anon function!!');
})

File: L18-Intro-to-Ruby/10-blocks.rb
def block_friendly_method
  puts "Starting..."
  yield # how do connect with other methods? (like callbacks in JS)
  puts "Finished!"
end

block_friendly_method do
  puts "Hello from anon block!!!"
end

block_friendly_method do
  message = gets.chomp
  puts "message: #{message}"
end